packages/devtools-proxy-support/src/ssh.ts

```diff
@@ -4,14 +4,28 @@ import type { DevtoolsProxyOptions } from './proxy-options';
 import type { AgentWithInitialize } from './agent';
 import type { ClientRequest } from 'http';
 import type { Duplex } from 'stream';
-import type { ClientChannel, ConnectConfig } from 'ssh2';
-import { Client as SshClient } from 'ssh2';
+import type { ClientChannel, ConnectConfig, Client as SshClient } from 'ssh2';
 import EventEmitter, { once } from 'events';
 import { promises as fs } from 'fs';
 import { promisify } from 'util';
 import type { ProxyLogEmitter } from './logging';
 import { connect as tlsConnect } from 'tls';
 import type { Socket } from 'net';
+import { getFips } from 'crypto';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+function ssh2(): typeof import('ssh2') {
+  if (getFips()) {
+    // ssh2 uses a WASM implementation of the non-FIPS-compliant Poly1305 hash algorithm
+    throw new Error(
```

Comment:
I might be wrong, but I feel this is the kind of error that we will find somewhere in a JIRA ticket and we will need to add logs later to understand a bit more. Is there any useful information we can add to the exception message?

Reply:
Hm yeah, I've had the same thought but haven't really come to a conclusion yet. I've added the package name here in 388c27a3851d5ac733a3e4099572a9c262acd25e, that makes it hopefully a bit easier to track down if this ever comes up.

Currently, I'm not too worried, because:
- SSH support is only explicitly exposed in Compass
- FIPS support is only explicitly exposed in mongosh

so at least in the current world, we shouldn't expect too much trouble coming out of this.

Reply:
That sounds good enough for me! Is there any situation (don't know much how FIPS work) where a user could unknowingly activate FIPS support or an option to tell them why FIPS is activated? If not (to any of the questions haha) just feel free to resolve this comment and merge.

Reply:
For mongosh, they'd typically either specify `--tlsFIPSMode` explicitly or call `crypto.setFips(1)` themselves.

I _guess_ there's a chance that this can be configured through the system OpenSSL config, but honestly, I don't think we've ever had any issues of that kind reported to us